connector/saml/saml.go

```diff
@@ -334,19 +353,97 @@ func (p *provider) HandlePOST(s connector.Scopes, samlResponse string) (ident co
 	return ident, nil
 }
 
+func (p *provider) validateStatus(resp *response) error {
+	status := resp.Status
+	if status == nil {
+		return fmt.Errorf("response did not contain a Status")
+	}
+	statusCode := status.StatusCode
+	if statusCode == nil {
+		return fmt.Errorf("response did not contain a StatusCode")
+	}
+	if statusCode.Value != "urn:oasis:names:tc:SAML:2.0:status:Success" {
+		return fmt.Errorf("status code of the response was not Success")
+	}
+	return nil
+}
+
+func (p *provider) validateSubjectConfirmation(subject *subject) error {
```

Comment:
Can we add some notes that point to where this is discussed in the spec? It seems arbitrary otherwise.